### PR TITLE
PLT-3140 Add SAML to System Console

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Added SAML Settings to System Console

#### Issue/Ticket Link
https://mattermost.atlassian.net/browse/PLT-3140

#### Related PR
https://github.com/mattermost/mattermost-driver-javascript/pull/14
https://github.com/mattermost/enterprise/pull/37

### CODE REVIEW

File: webapp/utils/web_client.jsx [hunk 1]
```diff
@@ -95,6 +95,24 @@ class WebClientClass extends Client {
             return success(res.body);
         });
     }
+
+    uploadCertificateFile(file, success, error) {
```
Comment (crspeller) [inline webapp/utils/web_client.jsx]: Shouldn't these be in client.jsx?

Comment (enahum) [inline webapp/utils/web_client.jsx]: Dunno there just to upload the files with system console, do we want this ability in the client.jsx? if that the case happy to move it

**Comment by jwilander** [comment]
+1 other than CS's comment

**Comment by enahum** [comment]
OK I'll move the client requests to client.jsx

**Comment by jasonblais** [comment]
Tested SAML on a fresh instance in webrtc.mattermost.com, testing Okta configuration on Windows 10, as well as searched for potential corner cases.

**UI Error Message Text**

Click `Save` in **System Console > Authentication > SAML** with `Enable Login with SAML` set to `true`
1. When `SAML SSO URL` is blank
   - observed: `Identity Provider Single Sign-On URL must be a valid URL and start with http:// or https://.` error message
   - expected: `SAML SSO URL must be a valid URL and start with http:// or https://.` error message

'
2. When `Identity Provider Issuer URL` is blank,
   - observed: `Identity Provider Descriptor URL must be a valid URL and start with http:// or https://.` error message
   - expected: `Identity Provider Issuer URL must be a valid URL and start with http:// or https://.` error message

'
3. When `Identity Provider Public Certificate` is blank,
   - observed: `Idp Certificate File missing in SAML Settings` error message
   - expected: `Identity Provider Public Certificate is missing.` error message

It appears the relevant error messages weren't updated with the new System Console titles. So I won't add the other cases, but let me know if you have any questions

**Functionality**: IDP set in Okta with encryption
4. Enable SAML login in the System Console with `SAML SSO URL`, `Identity Provider Issuer URL` and `Identity Provider Public Certificate` fields entered. Go to **Account Settings > Security > Sign-in Method** and attempt to turn on SAML SSO as usual
   - observed: Sign-in method switch is unsuccssful and I'm brought to this URL with a blank screen https://webrtc.spinmint.com/claim/email_to_oauth?action=email_to_sso&email=jason@mattermost.com
   - expected: Switch should be successful. When attempting to login with SAML receive an error message `SAML login was unsuccessful because encryption is not enabled. Please contact your administration.` (I think ?)
   

Docs in process here: https://github.com/mattermost/docs/tree/saml

**Comment by jwilander** [comment]
Also once the driver changes go in you'll need to point `webapp/package.json` at them

**Comment by jasonblais** [comment]
+1, tested latest changes with Elias on webrtc.spinmint.com

**Pull request merged**